qiskit_ibm_provider/job/ibm_circuit_job.py

```diff
@@ -289,65 +271,26 @@ def cancel(self) -> bool:
             ``True`` if the job is cancelled, else ``False``.
 
         Raises:
-            IBMJobApiError: If an unexpected error occurred when communicating
-                with the server.
+            IBMJobInvalidStateError: If the job is in a state that cannot be cancelled.
+            IBMJobError: If unable to cancel job.
         """
         try:
-            response = self._api_client.job_cancel(self.job_id())
-            self._cancelled = get_cancel_status(response)
+            self._runtime_client.job_cancel(self.job_id())
+            self._cancelled = True
             logger.debug(
-                'Job %s cancel status is "%s". Response data: %s.',
+                'Job %s cancel status is "%s".',
                 self.job_id(),
                 self._cancelled,
-                response,
             )
+            self._ws_client.disconnect(WebsocketClientCloseCode.CANCEL)
+            self._status = JobStatus.CANCELLED
```

Comment:
I'm pretty sure disconnecting with a cancel code doesn't actually cancel the job. It's only used to tell ws client not to re-connect if the job was cancelled. You actually need to call the cancel endpoint to cancel.

Reply:
On line 278 we're calling the endpoint - `self._runtime_client.job_cancel(self.job_id())`